### DISCUSSION TITLE
Allow type definitions for $listeners property.

### DISCUSSION THREAD

**Discussion by NiekNijland** [livewire/livewire#4732] [3. Feature ideas]
![image](https://user-images.githubusercontent.com/43474728/155701208-a51a781e-aa91-4ef0-9542-feda89c50e58.png)

As stated in title.

**Comment by silnarm** [comment]
Currently Livewire supports PHP 7.2.5+
Typed properties were introduced in 7.4

So.... soon perhaps, but not yet :)

Of course this potentially raises the question of why the latest Livewire versions support PHP versions that are themselves no longer supported.

**Reply by joshhanley** [reply]
@silnarm when Livewire V2 was released, it was released when Laravel 7 was the main Laravel version which had support for PHP 7.2.5+. So Livewire just followed that.

This will change in the next major version of Livewire 🙂

**Reply by silnarm** [reply]
Perfectly sensible. Never would have occurred to me ;-)

**Comment by joshhanley** [comment]
@NiekNijland yeah @silnarm is correct, this will have to wait for V3 as it wouldn't currently work due to Livewire supporting PHP 7.2.
